File: BE_sprint2/sprint/src/main/java/com/example/sprint/model/Product.java
package com.example.sprint.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private Double price;
    private Integer amountExist;
    @Column(columnDefinition = "TEXT")
    private String description;
    private String code;
    private String size;
    @ManyToOne
    private Port port;
    @ManyToOne
    private ButtonSupport buttonSupport;
    @ManyToOne
    private FrequencyBand frequencyBand;
    @ManyToOne
    private TypeDevice typeDevice;
    @ManyToOne
    private Anteing anteing;
    @ManyToOne
    private TypeAnteing typeAnteing;
    @ManyToOne
    private SpeedWifi speedWifi;
    @ManyToOne
    private StandardNetwork standardNetwork;
    @ManyToOne
    private UserConnect userConnect;
    @ManyToOne
    private CoverageDensity coverageDensity;
    @ManyToOne
    private Manufacture manufacture;
    @ManyToOne
    private MadeIn madeIn;
    @ManyToOne
    private Guarantee guarantee;
    @ManyToOne
    private Promotion promotion;
    @OneToMany(mappedBy = "product")
    @JsonBackReference
    private List<OrderDetail> orderDetailList;
    @OneToMany(mappedBy = "product")
    private List<ImageProduct> imageProducts;
    @Column(columnDefinition = "bit")
    private boolean flagDeleted = false;
    @Column(columnDefinition = "bit")
    private boolean flagPromoted = false;
}

File: FE_sprint2/sprint/src/app/model/interface/product.ts
import {ImageProducts} from './image-products';
import {Manufacture} from './manufacture';

interface Port {
  id?: number;
  name?: string;
}

interface ButtonSupport {
  id?: number;
  name?: string;
}


interface FrequencyBand {
  id?: number;
  name?: string;
}

interface TypeDevice {
  id?: number;
  name?: string;
}

interface Anteing {
  id?: number;
  amount?: number;
}

interface TypeAnteing {
  id?: number;
  name?: string;
}

interface SpeedWifi {
  id?: number;
  name?: string;
}

interface StandardNetwork {
  id?: number;
  name?: string;
}

interface UserConnect {
  id?: number;
  name?: string;
}

interface CoverageDensity {
  id?: number;
  name?: string;
}

interface MadeIn {
  id?: number;
  name?: string;
}

interface Guarantee {
  id?: number;
  name?: string;
}

interface Promotion {
  id?: number;
  percentPromotion?: number;
}


export interface Product {
  id?: number;
  name?: string;
  price?: number;
  description?: string;
  size?: string;
  port?: Port;
  buttonSupport?: ButtonSupport;
  frequencyBand?: FrequencyBand;
  typeDevice?: TypeDevice;
  anteing?: Anteing;
  typeAnteing?: TypeAnteing;
  speedWifi?: SpeedWifi;
  standardNetwork?: StandardNetwork;
  userConnect?: UserConnect;
  coverageDensity?: CoverageDensity;
  manufacture?: Manufacture;
  madeIn?: MadeIn;
  guarantee?: Guarantee;
  promotion?: Promotion;
  imageProducts?: ImageProducts[];
}
